alibi_detect/od/knn.py

```diff
@@ -0,0 +1,56 @@
+from typing import Callable, Literal, Union, Optional
+import numpy as np
+import os
+
+from alibi_detect.od.base import OutlierDetector
+from alibi_detect.od.aggregation import BaseTransform
+
+from alibi_detect.od.backends import KNNTorch, KNNKeops
+from alibi_detect.utils.frameworks import BackendValidator
+from alibi_detect.od.config import ConfigMixin
+from alibi_detect.saving.registry import registry
+
+X_REF_FILENAME = 'x_ref.npy'
+
+backends = {
+    'pytorch': KNNTorch,
+    'keops': KNNKeops
+}
+
+@registry.register('KNN')
+class KNN(OutlierDetector, ConfigMixin):
+    CONFIG_PARAMS = ('k', 'kernel', 'aggregator', 'normaliser', 'backend')
```

Comment:
Wondering if we can mostly automate this by setting it in the constructor using `locals()` (with appropriate exclusion rules). Or perhaps this is good enough if we have appropriate tests to ensure everything we want to (and don't want to) serialize is done for every detector. To be fair, in either case we'll have to do some manual lists of arguments to include/exclude so having it at the top as a class attribute at least makes it fully visible.

Reply:
On further inspection I see that there is a standardized call `self_set_config(locals())` which then looks at `CONFIG_PARAMS` to decide what to do with each argument. I think this is a good approach overall.